src/Marten/Services/IdentityMap.cs

```diff
@@ -9,18 +11,25 @@ namespace Marten.Services
     public class IdentityMap : IIdentityMap
     {
         private readonly ISerializer _serializer;
+        private readonly IEnumerable<IDocumentSessionListener> _listeners;
 
         private readonly Cache<Type, ConcurrentDictionary<object, object>> _objects
             = new Cache<Type, ConcurrentDictionary<object, object>>(_ => new ConcurrentDictionary<object, object>());
 
-        public IdentityMap(ISerializer serializer)
+        public IdentityMap(ISerializer serializer, IEnumerable<IDocumentSessionListener> listeners)
         {
             _serializer = serializer;
+            _listeners = listeners?.Any() == true ? listeners : null;
```

Comment:
@tim-cools I'd rather we used a "nullo" empty list here instead of having to remember to always use the "?" operator. Nit pick-y, I know.